glean-core/src/database/mod.rs

```diff
@@ -104,6 +104,30 @@ fn file_size(path: &Path) -> Option<NonZeroU64> {
         .and_then(NonZeroU64::new)
 }
 
+/// Determine the size of all the files in the given file directory.
+///
+///  # Arguments
+///
+///  -`path` - The path to the directory
+///
+///  # Returns
+///
+/// Returns the non-zero combined size of all files,
+/// or `None` on error or if the size is `0`.
+fn database_size(dir: &Path) -> Option<NonZeroU64> {
+    let mut total_size = 0;
+    if let Ok(entries) = fs::read_dir(dir) {
+        for entry in entries {
+            if let Ok(entry) = entry {
```

Comment:
An entry could be another directory.
We don't use nested directories here, so it _shouldn't happen. 

Let's still handle it by ignoring it explicitly. You can get an entries [`file_type`](https://doc.rust-lang.org/std/fs/struct.DirEntry.html#method.file_type) and check that `is_file` is true.